python/ray/actor.py

```diff
@@ -1377,12 +1376,6 @@ def exit_actor():
     """
     worker = ray._private.worker.global_worker
     if worker.mode == ray.WORKER_MODE and not worker.actor_id.is_nil():
-        # Intentionally disconnect the core worker from the raylet so the
-        # raylet won't push an error message to the driver.
-        ray._private.worker.disconnect()
-        # Disconnect global state from GCS.
-        ray._private.state.state.disconnect()
```

Comment:
Does this not have any unintended side effects? Looks like it could cause a behavior change for some workloads?

Reply:
This will exit, and then when exit happens, it will call the `Exit` API, which calls disconnect. This premature disconnect call seems to be the root cause of the segfault.